src/lib.rs

```diff
@@ -71,17 +71,68 @@
 //! rt = ["cortex-m-rt/device"]
 //! ```
 //!
-//! ## target != cortex-m
+//! ## target = msp430
 //!
-//! When the target is msp430, riscv or none `svd2rust` will emit only the `lib.rs` file. Like in
+//! MSP430 does not natively use the SVD format. However, SVD files can be generated using the
+//! [`msp430_svd` application](https://github.com/pftbest/msp430_svd). Most header and DSLite
+//! files provided by TI are mirrored in the repository of `msp430_svd`.
+//!
+//! When targeting the MSP430 architecture `svd2rust` will _also_ generate three files in the
+//! current directory:
+//!
+//! - `build.rs`, build script that places `device.x` somewhere the linker can find.
+//! - `device.x`, linker script that weakly aliases all the interrupt handlers to the default
+//! exception handler (`DefaultHandler`).
+//! - `lib.rs`, the generated code.
+//!
+//! All these files must be included in the same device crate. The `lib.rs` file contains several
+//! inlined modules and its not formatted. It's recommend to split it out using the [`form`] tool
+//! and then format the output using `rustfmt` / `cargo fmt`:
+//!
+//! [`form`]: https://crates.io/crates/form
+//!
+//! ``` text
+//! $ msp430gen msp430g2553 > out.svd
+//!
+//! $ xmllint -format out.svd > msp430g2553.svd
+//!
+//! $ svd2rust -i msp430g2553.svd
+//!
+//! $ rm -rf src
+//!
+//! $ form -i lib.rs -o src/ && rm lib.rs
+//!
+//! $ cargo fmt
+//! ```
+//!
+//! The resulting crate must provide an opt-in "rt" feature and depend on these crates:
+//! `bare-metal` v0.2.x, `msp430` v0.1.x, `msp430-rt` v0.2.x and `vcell` v0.1.x. Furthermore
+//! the "device" feature of `msp430-rt` must be enabled when the "rt" feature is enabled. The
+//! `Cargo.toml` of the device crate will look like this:
+//!
+//! ``` toml
+//! [dependencies]
+//! bare-metal = "0.2.0"
+//! msp430 = "0.1.0"
+//! vcell = "0.1.0"
+//!
+//! [dependencies.msp430-rt]
+//! optional = true
+//! version = "0.2.0"
+//!
+//! [features]
+//! rt = ["msp430-rt/device"]
```

Comment:
Haven't tested this but I think this needs to be added to svd2rust-regress

Reply:
Could you elaborate on what you mean by "adding this to sv2rust-regress"?

Reply:
@Emilgardis Okay, I think I have a grasp on this now.

I fixed the regression tests so the `msp430` test passes. As of `svd2rust` 17.0 plus this PR (which actually should probably be `18.0` since this is a breaking change), `svd2rust` will generate a PAC that is _not_ compatible with `msp430-rt v0.1.x`. Specifically:

* `INTERRUPTS` array has been renamed `__INTERRUPTS`; this matters to the linker script provided with `msp430-rt`.
* `interrupts!` macro replaces with attribute macro.

Right now, it seems nothing enables the `rt` feature of _any_ PACs (`lib.rs`) generated by the regression suite. Consequently, adding `msp430-rt v0.1.0` as a dep still passes- it's basically a no-op. Same situation would happen if I decided to publish and use `msp430-rt v0.2.0` as a prereq for this PR.

Since I'm going to be upgrading `msp430` packages in lockstep very soon, and it seems that `cortex-m` is also affected by not enabling the `rt` feature,  do you object if we hold off on adding this functionality to `svd2rust-regress` for now?

Reply:
Fine for me :+1: